src/unix/syscall.c

```diff
@@ -1697,6 +1697,14 @@ sysreturn fcntl(int fd, int cmd, int arg)
 
         f->flags = arg & ~O_CLOEXEC;
         return set_syscall_return(current, 0);
+    case F_GETLK:
+        if (arg) {
+            ((struct flock *)(long)arg)->l_type = F_UNLCK;
```

Comment:
I think perhaps arg should be changed to a u64 or void *; otherwise the address could possibly be truncated to 32 bits. (It probably passes in the runtime test below as the stack is in 32-bit space, but could fail for mmapped memory in high mem.)

Reply:
Thanks for the review!

I went with `u64` on this one after taking a quick look at the Linux implementation; I think it makes sense here too since `arg` its mostly used in numeric capacity rather than a pointer, so this would avoid excessive type casting.